cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -790,7 +814,25 @@ func (ht *hcsTask) Share(ctx context.Context, req *shimdiag.ShareRequest) error
 	if ht.host == nil {
 		return errTaskNotIsolated
 	}
-	return ht.host.Share(ctx, req.HostPath, req.UvmPath, req.ReadOnly)
+	// For hyper-v isolated WCOW the task used isn't the standard hcsTask so we
+	// only have to deal with the LCOW case here.
+	st, err := os.Stat(req.HostPath)
+	if err != nil {
+		return fmt.Errorf("could not open '%s' path on host: %s", req.HostPath, err)
+	}
+	var (
+		hostPath       string = req.HostPath
+		restrictAccess bool
+		fileName       string
+		allowedNames   []string
+	)
+	if !st.IsDir() {
+		hostPath, fileName = filepath.Split(hostPath)
+		allowedNames = append(allowedNames, fileName)
+		restrictAccess = true
+	}
+	_, err = ht.host.AddPlan9(ctx, hostPath, req.UvmPath, req.ReadOnly, restrictAccess, allowedNames)
+	return err
```

Comment:
why is this here?

Reply:
Good question.. rebasing madness. Fixing